osg_configure/configure_modules/gratia.py

```diff
@@ -118,71 +109,7 @@ def parse_configuration(self, configuration):
-                        continue
-                    lsf_location = configfile.Option(name='lsf_location',
-                                                     default_value='/usr/bin')
-                    configfile.get_option(configuration, 'LSF', lsf_location)
-                    self._probe_config['lsf'] = {'lsf_location': lsf_location.value}
-
-                    log_option = configfile.Option(name='log_directory',
-                                                   required=configfile.Option.OPTIONAL,
-                                                   default_value='')
-                    configfile.get_option(configuration, 'LSF', log_option)
-                    self._probe_config['lsf']['log_directory'] = log_option.value
-                elif probe == 'sge':
-                    if BaseConfiguration.section_disabled(configuration, 'SGE'):
-                        # if section is disabled then the following code won't work
-                        # since the parse_configuration will short circuit, so
-                        # give a warning and then move on
-                        self.log("Skipping SGE gratia probe configuration since SGE is disabled",
-                                 level=logging.WARNING)
-                        continue
-                    sge_config = SGEConfiguration(logger=self.logger)
-                    sge_config.parse_configuration(configuration)
-                    self._probe_config['sge'] = {'sge_accounting_file': sge_config.get_accounting_file()}
-                elif probe == 'slurm':
-                    if BaseConfiguration.section_disabled(configuration, 'SLURM'):
-                        # if section is disabled then the following code won't work
-                        # since the parse_configuration will short circuit, so
-                        # give a warning and then move on
-                        self.log("Skipping Slurm gratia probe configuration since Slurm is disabled",
-                                 level=logging.WARNING)
-                        continue
-                    slurm_config = SlurmConfiguration(logger=self.logger)
-                    slurm_config.parse_configuration(configuration)
-                    self._probe_config['slurm'] = {'db_host': slurm_config.get_db_host(),
-                                                    'db_port': slurm_config.get_db_port(),
-                                                    'db_user': slurm_config.get_db_user(),
-                                                    'db_pass': slurm_config.get_db_pass(),
-                                                    'db_name': slurm_config.get_db_name(),
-                                                    'cluster': slurm_config.get_slurm_cluster(),
-                                                    'location': slurm_config.get_location()}
-                elif probe == 'htcondor-ce':
+                if probe == 'htcondor-ce':
```

Comment:
Huh, didn't think `in` worked in iterators. Looks like it does though.

Reply:
Yeah, I think issues can arise if you check for an obj in the iterator that shows up once but that should be fine here since you don't use the same `probes_iter` elsewhere